website/docs/llm_configuration.ipynb

```diff
@@ -0,0 +1,346 @@
+    "\n",
+    "### Extra model client parameters\n",
+    "\n",
+    "It is also possible to passthrough parameters through to the OpenAI client. Parameters that correspond to the [`OpenAI` client](https://github.com/openai/openai-python/blob/d231d1fa783967c1d3a1db3ba1b52647fff148ac/src/openai/_client.py#L67) or the [`OpenAI` completions create API](https://github.com/openai/openai-python/blob/d231d1fa783967c1d3a1db3ba1b52647fff148ac/src/openai/resources/completions.py#L35) can be supplied.\n",
+    "\n",
+    "This is commonly used for things like `temperature`, or `timeout`.\n",
+    "\n",
+    "## Example\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "llm_config = {\n",
+    "    \"config_list\": [\n",
+    "        {\n",
+    "            \"model\": \"gpt-4\",\n",
+    "            \"api_key\": os.environ.get(\"AZURE_OPENAI_API_KEY\"),\n",
+    "            \"api_type\": \"azure\",\n",
+    "            \"base_url\": os.environ.get(\"AZURE_OPENAI_API_BASE\"),\n",
+    "            \"api_version\": \"2023-12-01-preview\",\n",
+    "        },\n",
+    "        {\n",
+    "            \"model\": \"llama-7B\",\n",
+    "            \"base_url\": \"http://127.0.0.1:8080\",\n",
+    "            \"api_type\": \"openai\",\n",
+    "        },\n",
+    "    ],\n",
+    "    \"temperature\": 0.9,\n",
+    "    \"timeout\": 300,\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "\n",
+    "## Other helpers for loading a config list\n",
+    "\n",
+    "- [`get_config_list`](/docs/reference/oai/openai_utils#get_config_list): Generates configurations for API calls, primarily from provided API keys.\n",
+    "- [`config_list_openai_aoai`](/docs/reference/oai/openai_utils#config_list_openai_aoai): Constructs a list of configurations using both Azure OpenAI and OpenAI endpoints, sourcing API keys from environment variables or local files.\n",
+    "- [`config_list_from_models`](/docs/reference/oai/openai_utils#config_list_from_models): Creates configurations based on a provided list of models, useful when targeting specific models without manually specifying each configuration.\n",
+    "- [`config_list_from_dotenv`](/docs/reference/oai/openai_utils#config_list_from_dotenv): Constructs a configuration list from a `.env` file, offering a consolidated way to manage multiple API configurations and keys from a single file.\n"
+   ]
```

Comment:
My goal was to remove those functions from the main documentation to try and not overwhelm the user - however you're right I shouldn't remove this info. I am going to move these sections into the API docs for these functions

Reply:
Moved the examples for config_list_from_dotenv into the docstring. There is a rendering issue at the moment for the SDK docs which I am going to look at next since it is affecting a lot of the SDK reference.

Let me know if this solution is good for you.

Reply:
To be honest i think it would make sense to have your documentation for the autogen docs site (which is a lot easier to digest), and have a separate notebook for usage keeping all of the examples. The documentation you removed used to live in `./notebooks` but was moved to docs a few weeks back, the reason why i made that notebook in the first place was because it was fairly common for people on the discord to post code for loading configs and asking how to fix it

Reply:
I like that suggestion a lot - I'll create a notebook for these advanced examples so we can keep those examples around and make sure these docs link out to that notebook for those examples

Reply:
@AaronWard let me know if the latest commit resolves your concerns. I added back the examples notebook as `config_loader_utility_functions.ipynb`, but I removed some of the explanation sections to reduce duplicating docs (around filter_dict etc). I linked to this notebook from the section on these other functions in the docs page this PR introduces.